File: kickkart-frontend/src/admin/api/AdminOrderService.js
import api from "../../api/axios";

export const getAllOrders = async (page = 0, size = 10) => {

    const response = await api.get("/admin/orders", {
        params: {
            page,
            size,
        },
    });

    return response.data;

};

export const getOrderById = async (id) => {

    const response = await api.get(`/admin/orders/${id}`);

    return response.data;

};

export const updateOrderStatus = async (id, status) => {

    const response = await api.put(
        `/admin/orders/${id}/status`,
        null,
        {
            params: {
                status,
            },
        }
    );

    return response.data;

};

File: kickkart-backend/src/main/java/com/kickkart/service/AdminOrderService.java
package com.kickkart.service;

import com.kickkart.dto.response.OrderResponse;
import com.kickkart.entity.OrderStatus;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

public interface AdminOrderService {

    Page<OrderResponse> getAllOrders(Pageable pageable);

    OrderResponse getOrderById(Long id);

    OrderResponse updateOrderStatus(Long id, OrderStatus status);

}